src/tools/tidy/src/style.rs

```diff
@@ -389,10 +401,12 @@ pub fn check(path: &Path, bad: &mut bool) {
         let mut lines = 0;
         let mut last_safety_comment = false;
         let mut comment_block: Option<(usize, usize)> = None;
-        let is_test = file.components().any(|c| c.as_os_str() == "tests");
+        let is_test = file.components().any(|c| c.as_os_str() == "tests")
+            || file.file_stem().unwrap() == "tests";
         // scanning the whole file for multiple needles at once is more efficient than
         // executing lines times needles separate searches.
-        let any_problematic_line = problematic_regex.is_match(contents);
+        let any_problematic_line =
+            filename != "style.rs" && !is_test && problematic_regex.is_match(contents);
```

Comment:
Why changing any logic except for regexes? Any file in repo can be named as `style.rs`.

Reply:
Correct, but the check, in which the regex checking takes place checks for the file-name too.

This change was made only to look for regex matches before the loop just in case it's needed.

If we replace it we should replace it in both places (which is a good idea imo)

Reply:
Anyway, fixed at 7404f08